tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```
@@ -0,0 +1,384 @@
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.slf4j.LoggerFactory.getLogger;
+import static java.nio.charset.Charset.defaultCharset;
+
+/**
+ * S3 client based on the Amazon AWS client API
+ */
+public class S3Client
+{
+    private static final Logger logger = getLogger(S3Client.class);
+
+    public static final String CONF_S3_AUTH_KEY_ID = "s3.auth.keyid";
+    public static final String CONF_S3_AUTH_SECRET = "s3.auth.secret";
+    public static final String CONF_S3_BUCKET_NAME = "s3.bucketname";
+
+    private final String bucketName;
+    private final AmazonS3 s3client;
+
+    /**
+     * Constructor
+     *
+     * @param keyId         AWS access key ID
+     * @param secret        AWS secret access key
+     * @param bucketName    S3 bucket to work in
+     */
+    public S3Client(
+            @Named(CONF_S3_AUTH_KEY_ID) String keyId,
+            @Named(CONF_S3_AUTH_SECRET) String secret,
+            @Named(CONF_S3_BUCKET_NAME) String bucketName)
+    {
+        // Check null values and set object variables
+        checkNotNull(keyId, "aws api access key id is null");
+        checkNotNull(secret, "aws api secret access key is null");
+        this.bucketName = checkNotNull(bucketName, "s3 bucket name is null");
+
+        // Initialize AWS SDK credentials
+        BasicAWSCredentials credentials = null;
+        try {
```

extract `BasicAWSCredential buildAWSCredentials(keyId, secret)` to hide exception handling inside it